lib/resources/git_resource.py

```diff
@@ -27,90 +37,181 @@ def __init__(self, database: DbConnector):
         self._database = database
 
     @staticmethod
-    def clone_repository(technology, url, path):
+    def clone_repository(url, path) -> Repo:
         """
         This method tries to clone the repository from the provided url.
         """
         logger.debug(f"Cloning repository {url} ...")
-        Git(path).clone(url)
-        repository = Repo(f"{path}/{technology}")
-        return repository
+        return Repo.clone_from(url, path, bare=True)
 
     @staticmethod
-    def get_tags(repository: Repo):
+    def get_hash(file_content) -> str or None:
         """
-        This method returns the list of tags sorted by commit date.
+        This method computes the hash of the provided file and returns it.
         """
-        return sorted(repository.tags, key=lambda tag: tag.commit.committed_datetime)
+        if file_content is None:
+            return None
+        return hashlib.sha256(file_content).hexdigest()
 
     @staticmethod
-    def get_hash(file):
+    def get_all_files_from_commit(commit: Commit) -> List[Tuple[FilePath, BlobHash]]:
         """
-        This method computes the hash of the provided file and returns it.
+        This method retrieves all files with their blob hash in a commit.
+        """
+        file_list = []
+
+        for blob in commit.tree.traverse():
+            if S_ISDIR(blob.mode):
+                continue
+            file_list.append((blob.path, blob.hexsha))
+        return file_list
+
+    def _hash_files(
+        self,
+        files: List[Tuple[FilePath, TagName, BlobHash]],
```

Comment:
In order to increase readability, you could use a named dictionary or a dataclass to name objects represented with `Tuple[FilePath, TagName, BlobHash]` and `Tuple[FilePath, TagName, FileHash]` as these two structures are used multiple times.